src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -683,6 +683,50 @@ struct accel_deadlock_status
   }
 };
 
+/*
+* Versal devices with PCIe interface program sc using xgq's
+* Transaction is initiated by writing to sysfs entry
+* This function throws different exceptions for 1.sysfs entry not present 2.sysfs access failure
+* Current sysfs access returns error string for both sysfs entry not present and sysfs access error
+*/
+struct program_sc
+{
+  using value_type = query::program_sc::value_type;
+  using result_type = query::program_sc::value_type;
+  static const std::string subdev;
+  static const std::string entry;
+
+  static result_type
+  get(const xrt_core::device* device, key_type key)
+  {
+    std::string err;
+    value_type value;
+    get_pcidev(device)->sysfs_get(subdev, entry, err, value, static_cast<value_type>(-1));
+    if (!err.empty())
+      throw xrt_core::query::sysfs_error(err);
+
+    return value;
+  }
+
+  static void
+  put(const xrt_core::device* device, key_type key, const boost::any& val)
+  {
+    auto pdev = get_pcidev(device);
+    //check for exsistence of sysfs node
+    std::string sysfs_path = pdev->get_sysfs_path(subdev, entry);
```

Comment:
is this function returns empty if path doesn't exist?

Reply:
Yes @chvamshi-xilinx, it returns empty string if path doesnot exist